math/mathmore/src/MCParameters.cxx

```diff
@@ -70,15 +69,15 @@ namespace Math {
       return *this;
    }
 
-   IOptions * VegasParameters::operator() () const {
+   std::unique_ptr<ROOT::Math::IOptions> VegasParameters::operator() () const {
       // convert to options (return object is managed by the user)
       GenAlgoOptions * opt = new GenAlgoOptions();
       opt->SetRealValue("alpha",alpha);
       opt->SetIntValue("iterations",iterations);
       opt->SetIntValue("stage",stage);
       opt->SetIntValue("mode",mode);
       opt->SetIntValue("verbose",verbose);
-      return opt;
+      return std::unique_ptr<ROOT::Math::IOptions>(opt);
```

Comment:
```suggestion
      return opt;
```
To *really* avoid manual memory management with operator `new`, it would be better to directly create `opt` as a unique_ptr and then simply return it.

And then further up, this  would mean doing:
```c++
auto opt = std::make_unique<GenAlgoOptions>();
```